src/tree/updater_gpu_hist.cu

```diff
@@ -1243,277 +1446,41 @@ class GPUHistMakerSpecialised{
       this->InitDataOnce(dmat);
       monitor_.StopCuda("InitDataOnce");
     }
-
-    column_sampler_.Init(info_->num_col_, param_.colsample_bynode,
```

Comment:
This ColumnSampler (in the GPUHistMakerSpecialised) never gets initialized. As the result, colsample_by* parameters are not passed to the shards.